apps/meteor/client/views/teams/contextualBar/channels/TeamsChannelsWithData.tsx

```diff
@@ -0,0 +1,92 @@
+import type { IRoom } from '@rocket.chat/core-typings';
+import { useMutableCallback, useLocalStorage, useDebouncedValue, useEffectEvent } from '@rocket.chat/fuselage-hooks';
+import { useSetModal, usePermission } from '@rocket.chat/ui-contexts';
+import type { FC, SyntheticEvent } from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+
+import { useRecordList } from '../../../../hooks/lists/useRecordList';
+import { AsyncStatePhase } from '../../../../lib/asyncState';
+import { roomCoordinator } from '../../../../lib/rooms/roomCoordinator';
+import CreateChannelWithData from '../../../../sidebar/header/CreateChannel';
+import { useRoom } from '../../../room/contexts/RoomContext';
+import { useRoomToolbox } from '../../../room/contexts/RoomToolboxContext';
+import RoomInfo from '../../../room/contextualBar/Info';
+import AddExistingModal from './AddExistingModal';
+import TeamsChannels from './TeamsChannels';
+import { useTeamsChannelList } from './hooks/useTeamsChannelList';
+
+const useReactModal = (Component: FC<any>, teamId: string, reload: () => void) => {
+	const setModal = useSetModal();
+
+	return useMutableCallback((e: SyntheticEvent) => {
+		e.preventDefault();
+
+		const handleClose = () => {
+			setModal(null);
+			reload();
+		};
+
+		setModal(() => <Component onClose={handleClose} teamId={teamId} />);
+	});
+};
```

Comment:
Should it be reusable? If so, move it to a new module.

Reply:
I don't like this approach, so I dropped it